test/metabase/driver/sql/query_processor_test.clj

```diff
@@ -1130,3 +1130,79 @@
     (binding [driver/*compile-with-inline-parameters* true]
       (is (= ["SELECT * FROM \"venues\" WHERE \"venues\".\"name\" = [my-string]"]
              (sql.qp/format-honeysql ::inline-value-test honeysql))))))
+
+(deftest ^:parallel expression-with-same-name-as-unused-column-test
```

Comment:
I ran into a new bug #46029 while working on this and fixed it as part of this PR